### ISSUE TITLE
Product Management: Unable to upload an image of the same name

### ISSUE THREAD

**Issue by masarapmabuhay** [usbong/UsbongStore#72]
Hi Zent,
I hope you are doing well.

I noticed that in the Product Management component, I am unable to upload an image with the same name as the the image I uploaded earlier for that same product.

The workaround I discovered is to return to the Manage page, click again the "Add Product" button, and upload the new image that has the same name.

Do you agree that it would be better to simply overwrite the earlier image with the new one upon upload?

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Hi Mike,

You should be able to overwrite an image using the edit feature regadless
of the image name.

I ll see if i can recreate ur scenario early this coming week.

I suspect though that this happened:
(1) old image was uploaded using a different means (direct upload)
(2) new image was uploaded using product management ui

The username or permission settings used by (1) n (2) were different. In
which case, we simply need to check ur directory and file permissions.

I ll check in detail early this week.

Thanks!


On Mar 4, 2018 11:03 AM, "Masarap Mabuhay" <notifications@github.com> wrote:

Hi Zent,
I hope you are doing well.

I noticed that in the Product Management component, I am unable to upload
an image with the same name as the the image I uploaded earlier for that
same product.

The workaround I discovered is to return to the Manage page, click again
the "Add Product" button, and upload the new image that has the same name.

Do you agree that it would be better to simply overwrite the earlier image
with the new one upon upload?

Thank you.

Peace,
Mike

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/usbong/UsbongStore/issues/72>, or mute the thread

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

OK!

Peace,
Mike

**Comment by zent-lim** [comment]
Afternoon Mike,

I'll look into this i the evening.

**Comment by zent-lim** [comment]
Hi Mike,

I was able to overwrite just fine.

==========
STEPS
==========

(1) I created a product using add product feature:

https://store.usbong.ph/index.php/manage/add

(2) I checked if image was written to expected directory:

zent@zent-K55VD:/var/www/html/UsbongStore/usbong_store/assets/images/books$
ls -la bianca.jpg
-rw-r--r-- 1 www-data www-data 13962 Mar  5 15:55 bianca.jpg

(3) I used the edit feature to update the image:

https://store.usbong.ph/index.php/manage/edit/437

Note: I uploaded the same image so filename is the same

(4) Using the time stamp, I checked if the image was uploaded correctly:

zent@zent-K55VD:/var/www/html/UsbongStore/usbong_store/assets/images/books$
ls -la bianca.jpg
-rw-r--r-- 1 www-data www-data 13962 Mar  5 15:56 bianca.jpg

Note: Time stamp has changed from 1555 to 1556.

==========
QUESTION
==========

What made you say that the image was not overwritten?

==========
ANALYSIS AND RECOMMENDATION
==========

(1) You used the front end of the store to see if the image was updated.

If this what what you did, it's probably just the browser cache.

Try clearing your cache before viewing the page again.

(2) Something unexpected happened on upload while you were editing.

If this is the case, let's look at your lie server directory permissions.

Good afternoon!

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

I was under the impression that the image was not overwritten, because I had to rename the file to successfully upload the same image.

Alternatively, as I mentioned when I opened this issue: 

I had to return to the Manage page, click again the "Add Product" button, and upload the new image that has the same name.

Here are my responses to your recommendations:
1) Clearing the browser cache did not resolve the issue.
2) OK. The server directory permissions for toys_and_collectibles is: `drwxrwxr-x`

Please let me know if there's anything else you need.

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
HI Mike,

Thanks for confirming the following:
(1) It's not a browser cache issue.
(2) Dir permission for toys_and_collectibles is drwxrwxr-x.

============
REQUEST 01:
============

(1) Please go to toys_and_collectibles and do the ff:

ls -la

(2) Please tell me the filename of the problem image.

(3) I'd like to see the ff please:

    (a) user, group of the directory

    (b) user, group, and permissions of the files inside

    (c) user, group, and permissions of the problem file

============
REQUEST 02:
============

Are you receiving any kind of error messages?

Thanks lots!

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

1) OK.

2) All the images I uploaded using the Product Management interface
--> Example, "Hubsan H501S X4 RC Quadcopter Body Shell.jpg"

3) For these, I'll send the details to you via email.
--> Interestingly, though, the file permissions for the problem files are: `-rw-r--r--`
--> Based on this, it appears that except for the owner, other users cannot overwrite the files. 
--> They can only read them.

And no, there are no error messages in the Product Management interface.

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Hi Mike

I still need to know the user and group of The files. Not just the user n
group pernission;

(A) Request 1 a
(B) first 2 data of request 1 b and c

Could u send me the whole output of ls -la pls?

Thanks!

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for the reply.

I have sent you via email the data you requested.

Thank you.

Peace,
Mike

**Comment by zent-lim** [comment]
Thanks for sending me the results of the command Mike!

===========
UPDATE
===========

I could not recreate your problem still.

But I have a pretty good idea of why it's not working.

===========
QUESTION
===========

(1) Is the problem image the image below by any chance?

-rw-r--r--  1 root     root      8743 Mar  5 11:52 Hubsan H122D X4 Racing
Drone.jpg

(2) It's because the other images have correct owner and group settings
"www-data":

-rw-r--r--  1 www-data www-data  7533 Mar  4 03:34 Hubsan H107P X4 Plus.jpg
-rw-r--r--  1 www-data www-data  5211 Mar  6 06:06 Hubsan H501S X4 RC
Quadcopter Body Shell.jpg
-rw-r--r--  1 www-data www-data  5772 Mar  6 06:17 Hubsan H502C X4 Star Cam
Beginner Quadcopter.jpg

(3) If yes, it seems that the problem image was not uploaded using the
product management feature. Maybe it was uploaded by FTP or SSH using root
account.

===========
REQUEST:
===========

Please try the commands below:

(1) ssh to your live server using root

(2) Go to toys dir

cd UsbongStore/usbong_store/assets/images/toys_and_collectibles

(3) Update owner from root to www-data

sudo chown www-data Hubsan\ H122D\ X4\ Racing\ Drone.jpg

(4) Update group from root to www-data

sudo chgrp www-data Hubsan\ H122D\ X4\ Racing\ Drone.jpg

After this, I think you should be able to use the edit feature just fine.

===========
NEXT:
===========

(1) If this small case test passes, let's do the same for the other files
below:

-rw-r--r--  1 ubuntu   ubuntu   14753 Sep 19 06:35 black widow marvel
90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   15077 Sep 19 06:03 BTAS killer croc.jpg
-rw-r--r--  1 ubuntu   ubuntu   15557 Sep 19 06:04 BTAS Mr. Freeze.jpg
-rw-r--r--  1 ubuntu   ubuntu   15558 Sep 19 06:04 BTAS scarecrow.jpg
-rw-r--r--  1 ubuntu   ubuntu   16483 Sep 19 06:36 bullseye and electra.jpg
-rw-r--r--  1 ubuntu   ubuntu   13789 Sep 19 06:40 captain america art
print.jpg
-rw-r--r--  1 ubuntu   ubuntu   19516 Sep 19 06:05 captain america cards.jpg
-rw-r--r--  1 ubuntu   ubuntu   15148 Sep 19 06:41 captain marvel.jpg
-rw-r--r--  1 ubuntu   ubuntu   16941 Sep 19 06:42 Carnage.jpg
-rw-r--r--  1 ubuntu   ubuntu   27018 Dec 20 08:57 Chinese Classic Cartoons
(7plus1 films).jpg
-rw-r--r--  1 ubuntu   ubuntu   17052 Sep 19 06:42 cyclops and jean grey
marvel 90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   15675 Aug  3  2017 DAREDEVIL.jpg
-rw-r--r--  1 ubuntu   ubuntu   17956 Sep 19 06:43 DD marvel toybiz 90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   16150 Sep 19 06:43 deathlok.jpg
-rw-r--r--  1 ubuntu   ubuntu   15586 Sep 19 06:44 Falcon marvel 90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   16229 Sep 19 06:44 green goblin marvel
90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   17345 Sep 19 06:45 green goblin marvel
toybiz 90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   10647 Nov  7 05:30 Gundam Fix The Works of
Katoki Hajime.jpg
-rw-r--r--  1 ubuntu   ubuntu   22960 Nov 27 04:22 hajimete no hisashiburi
no otona no piano (Music Sheet).jpg
-rw-r--r--  1 ubuntu   ubuntu   15119 Sep 19 06:46 iron fist marvel 90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   12418 Aug  3  2017 Marvel Legends Wave 1
X-MEN BAF Juggernaut.jpg
-rw-r--r--  1 ubuntu   ubuntu   10923 Sep 19 06:46 metal gear liquid snake
action figure.jpg
-rw-r--r--  1 ubuntu   ubuntu   16119 Sep 19 06:47 Morbius.jpg
-rw-r--r--  1 ubuntu   ubuntu   19491 Nov 27 04:22 Mr Children yasashii
piano solo (Music Sheet).jpg
-rw-r--r--  1 ubuntu   ubuntu   22390 Dec 29 07:48 Out of Your Mind
Essential Listening from the Alan Watts Audio Archives.jpg
-rw-r--r--  1 ubuntu   ubuntu   24171 Dec 16 09:12 Paranoia.jpg
-rw-r--r--  1 ubuntu   ubuntu   12574 Dec 16 09:11 Polaris The Roleplaying
Game.jpg
-rw-r--r--  1 ubuntu   ubuntu   12605 Sep 19 06:06 punisher art print.jpg
-rw-r--r--  1 ubuntu   ubuntu   17210 Sep 19 06:47 Rogue.jpg
-rw-r--r--  1 ubuntu   ubuntu   13976 Sep 19 07:37 scarlet spider.jpg
-rw-r--r--  1 ubuntu   ubuntu   22380 Aug  7  2017 Scratchbuild Revolution
Hobbymate 12th Anniversary Edition.jpg
-rw-r--r--  1 ubuntu   ubuntu   16700 Sep 19 06:49 silver surfer.jpg
-rw-r--r--  1 ubuntu   ubuntu   14786 Aug  3  2017 Spider-man Marvel Toybiz
90s.jpg
-rw-r--r--  1 ubuntu   ubuntu   28133 Nov 12 03:58 Super Mario Series for
Easy Piano (Music Sheet).jpg
-rw-r--r--  1 ubuntu   ubuntu   18660 Sep 19 06:49 talking venom.jpg
-rw-r--r--  1 ubuntu   ubuntu   24639 Nov  7 05:47 Teenage Mutant Ninja
Turtles 32 Cards and 35 Stickers.jpg
-rw-r--r--  1 ubuntu   ubuntu   27507 Nov  7 05:49 Teenage Mutant Ninja
Turtles 32 Cards and 8 Designs.jpg
-rw-r--r--  1 ubuntu   ubuntu   29516 Nov  7 05:41 Teenage Mutant Ninja
Turtles 32 Punch-Out Stand-up Cards.jpg
-rw-r--r--  1 ubuntu   ubuntu   23549 Aug 12  2017 The Art of the Matrix.jpg
-rw-r--r--  1 ubuntu   ubuntu   23735 Nov  7 05:28 The Dark Eye Core
Rules.jpg
-rw-r--r--  1 ubuntu   ubuntu   20376 Aug 24  2017 The Legend of Zelda
Hyrule Historia.jpg
-rw-r--r--  1 ubuntu   ubuntu   27999 Nov 12 03:59 The Legend of Zelda
Series for Easy Piano (Music Sheet).jpg
-rw-r--r--  1 ubuntu   ubuntu   26887 Nov 27 04:23 The Lord of the Rings
Instrumental Solos (Music Sheet).jpg
-rw-r--r--  1 ubuntu   ubuntu   17588 Sep 19 06:50 US agent.jpg
-rw-r--r--  1 ubuntu   ubuntu   15975 Sep 19 06:06 Venom marvel toybiz
90s.jpg

(2) ssh to your live server using root

(3) Go to toys dir

cd UsbongStore/usbong_store/assets/images/toys_and_collectibles

(4) Update owner from root to www-data

sudo chown www-data *.jpg

(5) Update group from root to www-data

sudo chgrp www-data *.jpg

Evening!

**Comment by masarapmabuhay** [comment]
Hi Zent,
Thank you for your reply here on GitHub.

I have updated the code on prod,
and have reverified the issue.

It seems that the issue is now resolved.

This case is now closed.

Thank you.

Peace,
Mike
p.s. I deleted your comment of March 7, 2018 to my preceding reply, so that our products' server details will not be easily viewable. Having said this, I have saved your recommended commands on my local computer.
